docs/tutorials/controlling-wedo2-motor.md

```diff
@@ -0,0 +1,255 @@
+    sleep 1
+    gatttool -i hci0 -b A0:E6:F8:1E:58:57 --char-write -a 0x003d -n 01010100
+
+(You need to run this script with sudo, unless you already have root previleges)
+
+We see that it uses the gatttool command to send a sequence of 4 bytes to one specific
+handler (0x003d). The WeDO 2.0 has several handlers but until LEGO Education releases
+the promised SDK this is the only handler we "know" how to use:
+
+This is meaning of those 4 bytes:
+* the first byte defines the port (01 or 02)
+* the second byte defines the command (01 = motor speed)
+* the third byte defines the length of the following argument(s) (01)
+* the fouth byte is the argument, in this case the speed percentage
+
+To spin in one direction we send a positive value from 1 to 100 (or 01 to 64 in
+hexadecimal).
+To spin in the opposite direction we send a "negative" value from
+255 to 156 (or FF to 9C in hexadecimal).
+To stop the motor we set the speed as zero (00).
+Please note that for small speed values (less than 20%) the motor will not respond.
+
+
+## Python example
+
+To use pyhton with the WeDO 2.0 we need a BLE library. Unfortunately BLE
+support in python is still quite imature but there is at least one library that
+works in ev3dev - [link](https://bitbucket.org/OscarAcena/pygattlib) gattlib
+
+    sudo apt-get install pkg-config libboost-python-dev libboost-thread-dev \
+     libbluetooth-dev libglib2.0-dev python-dev
+
+    pip install gattlib
+
+This library is also used as an extension for a more known library, [link](https://pypi.python.org/pypi/PyBluez) pybluez
+so if you want a library for both bluetooth "Classic" and BLE this would be better:
+
+    pip install pybluez
+    pip install pybluez[ble]
+
+Unfortunately I couldn't make it work in my ev3dev system.
+
+Please note that it takes **a lot** of memory and around 2 hours to install gattlib.
+After some failures ("virtual memory exhausted: Cannot allocate memory") I finally
+succeeded extending my ev3dev swapfile to almost 1 GB (please first check if you have
+enough free space in your SD card):
+
+    robot@ev3dev:~$ sudo dd if=/dev/zero of=/swapfile1 bs=1024 count=917504
+    917504+0 records in
+    917504+0 records out
+    939524096 bytes (940 MB) copied, 442.332 s, 2.1 MB/s
+
+it will take 5 to 110 minutes to allocate space for such a big file
+
+    robot@ev3dev:~$ sudo mkswap /swapfile1
+    Setting up swapspace version 1, size = 917500 KiB
+    no label, UUID=55fcb430-451b-4699-955c-5754bf65999b
+
+    robot@ev3dev:~$ sudo swapon /swapfile1
+    swapon: /swapfile1: insecure permissions 0644, 0600 suggested.
+
+this is a temporary measure so we'll skip security warning
+
+    robot@ev3dev:~$ sudo swapon -s
+    Filename				Type		Size	Used	Priority
+    /dev/zram0                             	partition	98300	8188	16383
+    /swapfile1                             	file    	917500	0	-1
+
+We don't want to use the swapfile in memory (it will overflow) so we disable it:
```

Comment:
It will still overflow with both swap files enabled? I have never had to disable zram when adding an extra swap file. Maybe your experience was different?